config/v1/types_image.go

```diff
@@ -67,6 +68,14 @@ type ImageSpec struct {
 	// internal cluster registry.
 	// +optional
 	RegistrySources RegistrySources `json:"registrySources"`
+
+	// imageStreamImportMode controls the import mode behaviour of imagestreams.
+	// It can be set to `Legacy` or `PreserveOriginal`. If this value is specified,
+	// this setting is applied to all newly created imagestreams which do not have the
+	// value set.
+	// +openshift:enable:FeatureGate=ImageStreamImportMode
+	// +optional
+	ImageStreamImportMode imagev1.ImportModeType `json:"imageStreamImportMode,omitempty"`
```

Comment:
Corss importing like this can cause issues, can we copy the type and enum values over please?

Reply:
done, but also had a clarification here: https://github.com/openshift/api/pull/1928#discussion_r1639304179